In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
#For Read the DATA SET
dataset = pd.read_csv('car_price.csv')

In [5]:
#Overall Data in Dataset.
dataset.shape

(5512, 10)

In [6]:
#For Check the Data 10 Rows.
dataset.head()

Unnamed: 0                                car_name car_prices_in_rupee  \
0           0  Jeep Compass 2.0 Longitude Option BSIV          10.03 Lakh   
1           1            Renault Duster RXZ Turbo CVT          12.83 Lakh   
2           2                      Toyota Camry 2.5 G          16.40 Lakh   
3           3                       Honda Jazz VX CVT           7.77 Lakh   
4           4        Volkswagen Polo 1.2 MPI Highline           5.15 Lakh   

   kms_driven fuel_type transmission  ownership  manufacture   engine    Seats  
0  86,226 kms    Diesel       Manual  1st Owner         2017  1956 cc  5 Seats  
1  13,248 kms    Petrol    Automatic  1st Owner         2021  1330 cc  5 Seats  
2  60,343 kms    Petrol    Automatic  1st Owner         2016  2494 cc  5 Seats  
3  26,696 kms    Petrol    Automatic  1st Owner         2018  1199 cc  5 Seats  
4  69,414 kms    Petrol       Manual  1st Owner         2016  1199 cc  5 Seats

In [8]:
#For Check the Data 10 Rows.
dataset.head()

Unnamed: 0                                car_name car_prices_in_rupee  \
0           0  Jeep Compass 2.0 Longitude Option BSIV          10.03 Lakh   
1           1            Renault Duster RXZ Turbo CVT          12.83 Lakh   
2           2                      Toyota Camry 2.5 G          16.40 Lakh   
3           3                       Honda Jazz VX CVT           7.77 Lakh   
4           4        Volkswagen Polo 1.2 MPI Highline           5.15 Lakh   

   kms_driven fuel_type transmission  ownership  manufacture   engine    Seats  
0  86,226 kms    Diesel       Manual  1st Owner         2017  1956 cc  5 Seats  
1  13,248 kms    Petrol    Automatic  1st Owner         2021  1330 cc  5 Seats  
2  60,343 kms    Petrol    Automatic  1st Owner         2016  2494 cc  5 Seats  
3  26,696 kms    Petrol    Automatic  1st Owner         2018  1199 cc  5 Seats  
4  69,414 kms    Petrol       Manual  1st Owner         2016  1199 cc  5 Seats

In [9]:
#For Check the Data Types.
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5512 entries, 0 to 5511
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Unnamed: 0           5512 non-null   int64 
 1   car_name             5512 non-null   object
 2   car_prices_in_rupee  5512 non-null   object
 3   kms_driven           5512 non-null   object
 4   fuel_type            5512 non-null   object
 5   transmission         5512 non-null   object
 6   ownership            5512 non-null   object
 7   manufacture          5512 non-null   int64 
 8   engine               5512 non-null   object
 9   Seats                5512 non-null   object
dtypes: int64(2), object(8)
memory usage: 430.8+ KB


In [14]:
#Check missing Content
#isnull() is Use for check null value in Data Set.
#sum() is use for calculate all null values in dataset.
#below this code is use for value come from Percentage.
#(dataset.isnull().sum()/dataset.shape[0])*100
dataset.isnull().sum()

Unnamed: 0             0.0
car_name               0.0
car_prices_in_rupee    0.0
kms_driven             0.0
fuel_type              0.0
transmission           0.0
ownership              0.0
manufacture            0.0
engine                 0.0
Seats                  0.0
dtype: float64

In [15]:
#Blank Row remove using Dropna
#inplace=True is use for save the file
dataset.dropna(inplace=True)

In [8]:
# select column using []
dataset['car_name']

0       Jeep Compass 2.0 Longitude Option BSIV
1                 Renault Duster RXZ Turbo CVT
2                           Toyota Camry 2.5 G
3                            Honda Jazz VX CVT
4             Volkswagen Polo 1.2 MPI Highline
                         ...                  
5507                   BMW X1 sDrive 20d xLine
5508                     BMW M Series M4 Coupe
5509                Jaguar XF 2.2 Litre Luxury
5510                        BMW 7 Series 730Ld
5511                 BMW 5 Series 520d M Sport
Name: car_name, Length: 5512, dtype: object

In [9]:
dataset['car_name'][0]

'Jeep Compass 2.0 Longitude Option BSIV'

In [11]:
x = dataset['car_name'][0]
x.index(" ")

4

In [12]:
#Slicing 
x = dataset['car_name'][0]
x[:x.index(" ")]

'Jeep'

In [17]:
#we create customise function for entire data
#Company name 
def company_name(x):
    return x[:x.index(" ")]

In [18]:
#we create customise function for Car_name
def car_name(x):
    return x[x.index(" ")+1:]

In [15]:
#Company name
dataset["Company_name"] = dataset["car_name"].apply(company_name)

In [21]:
#Car name 
dataset['car_name'] = dataset["car_name"].apply(car_name)

In [25]:
#company car name colum adjust
column_order = ['Company_name'] + [col for col in dataset.columns if col != 'Company_name']
dataset= dataset[column_order]

In [28]:
#Unnamed column Delete
dataset.drop(columns=['Unnamed: 0'], inplace=True)

In [29]:
dataset.head()

Company_name                           car_name car_prices_in_rupee  \
0         Jeep  Compass 2.0 Longitude Option BSIV          10.03 Lakh   
1      Renault               Duster RXZ Turbo CVT          12.83 Lakh   
2       Toyota                        Camry 2.5 G          16.40 Lakh   
3        Honda                        Jazz VX CVT           7.77 Lakh   
4   Volkswagen              Polo 1.2 MPI Highline           5.15 Lakh   

   kms_driven fuel_type transmission  ownership  manufacture   engine    Seats  
0  86,226 kms    Diesel       Manual  1st Owner         2017  1956 cc  5 Seats  
1  13,248 kms    Petrol    Automatic  1st Owner         2021  1330 cc  5 Seats  
2  60,343 kms    Petrol    Automatic  1st Owner         2016  2494 cc  5 Seats  
3  26,696 kms    Petrol    Automatic  1st Owner         2018  1199 cc  5 Seats  
4  69,414 kms    Petrol       Manual  1st Owner         2016  1199 cc  5 Seats

car prise
Problems in car price
35,000 , 28.90 Lakh , 1.20 Crors. 

In [30]:
dataset['car_prices_in_rupee']

0       10.03 Lakh
1       12.83 Lakh
2       16.40 Lakh
3        7.77 Lakh
4        5.15 Lakh
           ...    
5507    28.90 Lakh
5508    64.90 Lakh
5509    13.75 Lakh
5510    29.90 Lakh
5511    31.90 Lakh
Name: car_prices_in_rupee, Length: 5512, dtype: object

In [31]:
#Comma Replace in Car Price.
dataset['car_prices_in_rupee'].str.replace(',',' ')

0       10.03 Lakh
1       12.83 Lakh
2       16.40 Lakh
3        7.77 Lakh
4        5.15 Lakh
           ...    
5507    28.90 Lakh
5508    64.90 Lakh
5509    13.75 Lakh
5510    29.90 Lakh
5511    31.90 Lakh
Name: car_prices_in_rupee, Length: 5512, dtype: object

In [32]:
dataset['car_prices_in_rupee'] = dataset['car_prices_in_rupee'].str.replace(',',' ')

In [33]:
#Convertion on Rupees function
def rupe_change(x):
    p = x.split(" ")
    try:
        if p[1] == "Lakh":
            return str(round(float(p[0]) * 100000, 1))
        elif p[1] == "Crore":
            return str(round(float(p[0]) * 10000000, 1))
    except:
        return x

In [34]:
dataset['car_prices_in_rupee'].apply(rupe_change)

0       1003000.0
1       1283000.0
2       1640000.0
3        777000.0
4        515000.0
          ...    
5507    2890000.0
5508    6490000.0
5509    1375000.0
5510    2990000.0
5511    3190000.0
Name: car_prices_in_rupee, Length: 5512, dtype: object

In [35]:
dataset['car_prices_in_rupee']= dataset['car_prices_in_rupee'].apply(rupe_change)

In [38]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5512 entries, 0 to 5511
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Company_name         5512 non-null   object
 1   car_name             5512 non-null   object
 2   car_prices_in_rupee  5389 non-null   object
 3   kms_driven           5512 non-null   object
 4   fuel_type            5512 non-null   object
 5   transmission         5512 non-null   object
 6   ownership            5512 non-null   object
 7   manufacture          5512 non-null   int64 
 8   engine               5512 non-null   object
 9   Seats                5512 non-null   object
dtypes: int64(1), object(9)
memory usage: 430.8+ KB


In [42]:
dataset['car_prices_in_rupee'].astype("float")

0       1003000.0
1       1283000.0
2       1640000.0
3        777000.0
4        515000.0
          ...    
5507    2890000.0
5508    6490000.0
5509    1375000.0
5510    2990000.0
5511    3190000.0
Name: car_prices_in_rupee, Length: 5512, dtype: float64

In [39]:
#Change data type into column Object to float
#Object datatype means String 
dataset['car_prices_in_rupee'] = dataset['car_prices_in_rupee'].astype("float")

In [ ]:
kms_driven

In [43]:
dataset['kms_driven']

0       86,226 kms
1       13,248 kms
2       60,343 kms
3       26,696 kms
4       69,414 kms
           ...    
5507    45,000 kms
5508    29,000 kms
5509    90,000 kms
5510    79,000 kms
5511    42,000 kms
Name: kms_driven, Length: 5512, dtype: object

In [ ]:
dataset['kms_driven'] = dataset['kms_driven'].str.replace(",","")
dataset['kms_driven'] = dataset['kms_driven'].str.replace("kms","")

In [57]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5512 entries, 0 to 5511
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Company_name         5512 non-null   object 
 1   car_name             5512 non-null   object 
 2   car_prices_in_rupee  5389 non-null   float64
 3   kms_driven           5512 non-null   int32  
 4   fuel_type            5512 non-null   object 
 5   transmission         5512 non-null   object 
 6   ownership            5512 non-null   object 
 7   manufacture          5512 non-null   int64  
 8   engine               5512 non-null   object 
 9   Seats                5512 non-null   object 
dtypes: float64(1), int32(1), int64(1), object(7)
memory usage: 409.2+ KB


In [58]:
# Convert the 'kms_driven' column data type from int32 to int64
dataset['kms_driven'] = dataset['kms_driven'].astype('int64')

In [59]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5512 entries, 0 to 5511
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Company_name         5512 non-null   object 
 1   car_name             5512 non-null   object 
 2   car_prices_in_rupee  5389 non-null   float64
 3   kms_driven           5512 non-null   int64  
 4   fuel_type            5512 non-null   object 
 5   transmission         5512 non-null   object 
 6   ownership            5512 non-null   object 
 7   manufacture          5512 non-null   int64  
 8   engine               5512 non-null   object 
 9   Seats                5512 non-null   object 
dtypes: float64(1), int64(2), object(7)
memory usage: 430.8+ KB


#Engine

In [60]:
dataset["engine"]

0       1956 cc
1       1330 cc
2       2494 cc
3       1199 cc
4       1199 cc
         ...   
5507    2995 cc
5508    1968 cc
5509    2755 cc
5510    2967 cc
5511    1991 cc
Name: engine, Length: 5512, dtype: object

In [62]:
dataset['engine'] = dataset['engine'].str.replace(" cc","")
dataset['engine'] = dataset['engine'].astype('int64')

In [63]:
dataset['engine']

0       1956
1       1330
2       2494
3       1199
4       1199
        ... 
5507    2995
5508    1968
5509    2755
5510    2967
5511    1991
Name: engine, Length: 5512, dtype: int64

#Seats

In [68]:
dataset['Seats']

0       5 Seats
1       5 Seats
2       5 Seats
3       5 Seats
4       5 Seats
         ...   
5507    7 Seats
5508    5 Seats
5509    5 Seats
5510    6 Seats
5511    5 Seats
Name: Seats, Length: 5512, dtype: object

In [69]:
dataset['Seats'] = dataset['Seats'].str.replace(" Seats","")
dataset['Seats'] = dataset['Seats'].astype('int64')

In [70]:
dataset['Seats']

0       5
1       5
2       5
3       5
4       5
       ..
5507    7
5508    5
5509    5
5510    6
5511    5
Name: Seats, Length: 5512, dtype: int64

#manifacturing Date

In [72]:
dataset['manufacture']

0       2017
1       2021
2       2016
3       2018
4       2016
        ... 
5507    2018
5508    2015
5509    2013
5510    2015
5511    2017
Name: manufacture, Length: 5512, dtype: int64

In [76]:
#pd.to_datetime(dataset['manufacture'])
dataset['manufacture'] = pd.to_datetime(dataset['manufacture'], format='%Y').dt.year

In [78]:
dataset['manufacture'] = dataset['manufacture'].astype('int64')


In [80]:
dataset['manufacture']

0       2017
1       2021
2       2016
3       2018
4       2016
        ... 
5507    2018
5508    2015
5509    2013
5510    2015
5511    2017
Name: manufacture, Length: 5512, dtype: int64

In [82]:
dataset.head()

Company_name                           car_name  car_prices_in_rupee  \
0         Jeep  Compass 2.0 Longitude Option BSIV            1003000.0   
1      Renault               Duster RXZ Turbo CVT            1283000.0   
2       Toyota                        Camry 2.5 G            1640000.0   
3        Honda                        Jazz VX CVT             777000.0   
4   Volkswagen              Polo 1.2 MPI Highline             515000.0   

   kms_driven fuel_type transmission  ownership  manufacture  engine  Seats  
0       86226    Diesel       Manual  1st Owner         2017    1956      5  
1       13248    Petrol    Automatic  1st Owner         2021    1330      5  
2       60343    Petrol    Automatic  1st Owner         2016    2494      5  
3       26696    Petrol    Automatic  1st Owner         2018    1199      5  
4       69414    Petrol       Manual  1st Owner         2016    1199      5